### PR TITLE
[16.0][ADD] stock_production_lot_expired_date

### PR DESCRIPTION
**Pull request by glitchov**
Addition to OCA

### CODE REVIEW

**Comment by glitchov** [comment]
Bad module name

**Comment by rousseldenis** [comment]
@glitchov It's an addition, not a migration

**Review by lmignon — CHANGES_REQUESTED** [review]
@glitchov  Can you add "ACSONE SA/NV" into the author's list and add copyright for acsone PLZ

File: stock_production_lot_expired_date/tests/test_stock_production_lot_expired_date.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+from odoo.tests.common import TransactionCase
+
+
+class TestStockLotExpirationDates(TransactionCase):
+    at_install = False
+    post_install = True
```
Comment (lmignon) [inline stock_production_lot_expired_date/tests/test_stock_production_lot_expired_date.py]: @glitchov Les tests ne doivent pas tourner en post_install.. Tu peux supprimer ces deux lignes

**Review by lmignon — APPROVED** [review]

**Review by FrancoMaxime — CHANGES_REQUESTED** [review]
as you're adding a new module, can you regroup all your commits into one, with a name like "[ADD] stock_production_lot_expired_date"

**Review by FrancoMaxime — APPROVED** [review]
LGTM, thanks for the correction

File: stock_production_lot_expired_date/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This module allow to facilitate the input of stock production lot expired dates:
```
Comment (baimont) [inline stock_production_lot_expired_date/readme/DESCRIPTION.rst]:
```suggestion
This module allows to facilitate the input of stock production lot expired dates:
```

File: stock_production_lot_expired_date/views/res_config_settings.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record model="ir.ui.view" id="stock_config_settings_form_view">
+        <field
+            name="name"
+        >stock.config.settings.form (in stock_production_lot_expiry)</field>
```
Comment (baimont) [inline stock_production_lot_expired_date/views/res_config_settings.xml]:
```suggestion
        >stock.config.settings.form (in stock_production_lot_expired_date)</field>
```

File: stock_production_lot_expired_date/tests/test_stock_production_lot_expired_date.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+                )
+                self.assertEqual(
+                    production_lot.alert_date,
+                    self.date_25 if date_must_change else False,
+                )
+                self.assertEqual(
+                    production_lot.removal_date,
+                    self.date_26 if date_must_change else False,
+                )
+                # Check the onchange not fails with no value
```
Comment (baimont) [inline stock_production_lot_expired_date/tests/test_stock_production_lot_expired_date.py]:
```suggestion
                # Check the onchange does not fail with no value
```

File: stock_production_lot_expired_date/views/res_config_settings.xml (hunk 3 above)
Comment (baimont) [inline stock_production_lot_expired_date/views/res_config_settings.xml]:
```suggestion
        >stock.config.settings.form (in stock_production_lot_expired_date)</field>
```

**Review by baimont — APPROVED** [review]

**Comment by lmignon** [comment]
@glitchov Can you take into account the last comments from @baimont ? Thanks in advance!

**Comment by glitchov** [comment]
@baimont Can you check please ?

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**